### PR TITLE
Switch Scaleway to salt.utils.cloud.bootstrap()

### PR DESCRIPTION
**Pull request by techhat**
@edouardb, could you please take a look?

### CODE REVIEW

File: salt/cloud/clouds/scaleway.py [hunk 1]
```diff
@@ -272,98 +272,11 @@ def __query_node_data(server_name):
         finally:
             raise SaltCloudSystemExit(str(exc))
 
-    ssh_username = config.get_cloud_config_value(
-        'ssh_username', server_, __opts__, default='root'
+    vm_['ssh_host'] = ip_address
```
Comment (edouardb) [inline salt/cloud/clouds/scaleway.py]: `vm_` doesn't exist should be `server_`, same thing for next occurrences

Comment (edouardb) [inline salt/cloud/clouds/scaleway.py]: `ip_address` is not defined should be assigned from `data['public_ip']['address']`

```
if data.get('public_ip') is not None:
  ip_address = data['public_ip']['address']
```

**Comment by edouardb** [comment]
Hi @techhat,

After fixing the code I run some test on the create function and I get the following error:

```
Error: There was a query error: Cannot deploy salt in a VM if the 'ssh_keyfile' setting is not set and there is no password set for the vm. Check your provider for the 'change_password' option.
```

Any idea why?

Thanks

**Comment by techhat** [comment]
@edouardb, fixed. Do you have any comments on `ssh_password` or `ssh_keyfile`? One or both needs to be specified, and I didn't see either in your code.

**Comment by techhat** [comment]
@edouardb check my other question to you, it should answer your question to me.

**Comment by edouardb** [comment]
Right! :smile: 
Both ssh_keyfile and ssh_password should be used with ssh_keyfile default.

I tried  to add add  `ssh_key: /path/to/key.priv` into my profile but still got the error `Cannot deploy salt in a VM if the 'ssh_keyfile' setting is not set and there is no password set for the vm. Check your provider for the 'change_password' option.`

File: salt/cloud/clouds/scaleway.py [hunk 2]
```diff
@@ -272,98 +272,11 @@ def __query_node_data(server_name):
         finally:
             raise SaltCloudSystemExit(str(exc))
 
-    ssh_username = config.get_cloud_config_value(
-        'ssh_username', server_, __opts__, default='root'
+    server_['ssh_host'] = data['public_ip']['address']
+    server_['ssh_password'] = config.get_cloud_config_value(
```
Comment (edouardb) [inline salt/cloud/clouds/scaleway.py]: server_['ssh_password'] = config.get_cloud_config_value(
     'ssh_password', server_, __opts__
)

server_['ssh_keyfile'] = config.get_cloud_config_value(
     'ssh_keyfile', server_, __opts__
)

**Comment by techhat** [comment]
@edouardb `ssh_key` or `ssh_keyfile`?

**Comment by edouardb** [comment]
@techhat  `ssh_keyfile`

**Comment by edouardb** [comment]
Line 209 and 384, `’provider’: server_[‘provider’] ` must be changed to `’driver’ : server_[‘driver’]`  to prevent salt to raise the following exception:

```
[ERROR   ] Failed to create VM XXX. Configuration value 'provider' needs to be set
Traceback (most recent call last):
  File "/root/salt/salt/cloud/__init__.py", line 1256, in create
    output = self.clouds[func](vm_)
  File "/root/salt/salt/cloud/clouds/scaleway.py", line 209, in create
    'provider': server_['provider'],
KeyError: 'provider'
```
and 

```
[ERROR   ] Failed to create VM XXX. Configuration value 'provider' needs to be set
Traceback (most recent call last):
  File "/root/salt/salt/cloud/__init__.py", line 1256, in create
    output = self.clouds[func](vm_)
  File "/root/salt/salt/cloud/clouds/scaleway.py", line 384, in create
    'provider': server_['provider'],
KeyError: 'provider'
```

**Comment by rallytime** [comment]
@edouardb Thanks for testing this. Those stacktraces should be fixed in #25575 that I've referenced above.

**Comment by rallytime** [comment]
Go Go Jenkins!

**Comment by techhat** [comment]
Thanks @rallytime. Please hold off on merging for a moment, @edouardb and I are still working on the ssh_keyfile issue.

**Comment by techhat** [comment]
I have been unable to reproduce this issue. In fact, without this code I can spin up a machine but not provision it, but with this code, the machine is provisioned.

**Pull request merged**